core/templates/dev/head/dashboard/Dashboard.js

```diff
@@ -73,6 +88,26 @@ oppia.controller('Dashboard', [
       $window.location = '/create/' + explorationId;
     };
 
+    $scope.openExploration = function(status, explorationId) {
+      if (status === 'private') {
+        $scope.showExplorationEditor(explorationId);
+      } else {
+        DashboardBackendApiService.fetchExplorationStats(explorationId).then(
+          function(response) {
+            $scope.explorationStats[explorationId] = response.data;
+          }, function(errorResponse) {
+            if (FATAL_ERROR_CODES.indexOf(errorResponse.status) !== -1) {
```

Comment:
@AllanYangZhou Just realized that there's a minor issue with this - how do I detect if the request failed because the server has taken too long to respond ? Currently I'm (wrongly) assuming that if the request's status codes is not in the list of fatal error codes then it might be due to the timeout but I think this is a very vague way of implementing this!
A quick google search lead me to http://stackoverflow.com/questions/17873961/determine-request-time-out-in-angular-http - but I'm not sure if any of the answers here address this situation well enough!
/cc @seanlip